qualifier.py

```diff
@@ -0,0 +1,80 @@
+"""
+Use this file to write your solution for the Summer Code Jam 2020 Qualifier.
+
+Important notes for submission:
+
+- Do not change the names of the two classes included below. The test suite we
+  will use to test your submission relies on existence these two classes.
+
+- You can leave the `ArticleField` class as-is if you do not wish to tackle the
+  advanced requirements.
+
+- Do not include "debug"-code in your submission. This means that you should
+  remove all debug prints and other debug statements before you submit your
+  solution.
+"""
+import datetime
+import typing
+import re
+
+
+class ArticleField:
+    """The `ArticleField` class for the Advanced Requirements."""
+
+    def __init__(self, field_type: typing.Type[typing.Any]):
+        pass
+
+
+class Article:
+    """The `Article` class you need to write for the qualifier."""
+
+    def __init__(
+        self, title: str, author: str, publication_date: datetime.datetime, content: str
+    ):
+        self.title = title
+        self.author = author
+        self.publication_date = publication_date
+        self.content = content
+
+    def __repr__(self):
+        return f'<Article title="{self.title}" author=\'{self.author}\' publication_date=\'{self.publication_date.isoformat("T")}\'>'
+
+    def __len__(self):
+        return len(self.content)
+
+    def short_introduction(self, n_characters):
+        test = self.content.split()
+        original = self.content[:n_characters].split()
+        if original[-1] not in test:
+            original.remove(original[-1])
+        return " ".join(original)
+
+    def most_common_words(self, n_words):
+        word_split = self.content.lower()
+        word_split = re.findall(r"[\w]*", word_split)
+        new_list = []
+        for word in word_split:
+            if word == "":
+                continue
+            else:
+                new_list.append(word)
+        my_dict = {}
+        for word in new_list:
```

Comment:
Technically, this is totally fine. But if you would like to improve this part, you could use [`collections.defaultdict`](https://docs.python.org/2/library/collections.html#collections.defaultdict), which saves you the hassle of checking whether the key is already present in the dict.
Again, `my_dict` is not the best variable name, something like `word_counter` would fit better. `my_dict` could be everything. For such a small function, that name is fine, but as functions grow, it can get messy if you have a lot of those names and always have to go back up and see where it is defined and for what it is used.

Example of using `collections.defaultdict`
```py
my_dict = defaultdict(int)  # defaults all keys as {key: 0}
for word in new_list:
    my_dict[word] += 1
```